crates/iota-data-ingestion-core/src/reader.rs

```diff
@@ -283,14 +287,18 @@ impl CheckpointReader {
                 checkpoint.checkpoint_summary.sequence_number,
                 self.current_checkpoint_number
             );
-            self.checkpoint_sender.send(checkpoint).await?;
+            self.checkpoint_sender.send(checkpoint).await.map_err(|_| {
+                IngestionError::Channel(
+                    "unable to send checkpoint to executor, receiver half dropped".to_owned(),
```

Comment:
Nit: closed seems more precise: https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#errors
```suggestion
                    "unable to send checkpoint to executor, receiver half closed".to_owned(),
```